storage/src/mmr/mutable.rs

```diff
@@ -242,10 +268,60 @@ impl<K: Array, V: Array, H: CHasher> MutableMmr<K, V, H> {
         proof.verify_range_inclusion(hasher, &digests, start_pos, end_pos, root_hash)
     }
 
-    /// Consume the store and return its log of operations, from which the state of the store can be
-    /// recovered.
-    pub fn to_log(self) -> Vec<Operation<K, V>> {
-        self.log
+    /// Consume the store and return a [StoreState] from which the current active state of the store
+    /// can be fully recovered and fully proven.
+    pub fn to_state(mut self) -> StoreState<K, V, H> {
+        self.prune_known_inactive();
+
+        let prune_to_pos = leaf_num_to_pos(self.pruned_loc);
+        let pinned_nodes = self.ops.node_digests_to_pin(prune_to_pos);
+
+        StoreState {
+            log: self.log.into(),
+            pruned_loc: self.pruned_loc,
+            pinned_nodes,
+        }
+    }
+
+    /// Raise the inactivity floor, stopping after `max_steps` steps or when there are no more
+    /// inactive operations. With each step, either an inactive operation is advanced over, or an
+    /// active operation is re-applied to the tip and then advanced over.
+    ///
+    /// This method does not change the state of the store's snapshot, but it may change the root
+    /// hash of the store if some step re-applies an active operation.
+    #[allow(dead_code)] // TODO: Remove when method gets used.
+    pub fn raise_inactivity_floor(&mut self, hasher: &mut H, max_steps: u64) {
+        let active_ops = self.snapshot.len() as u64;
+        for _ in 0..max_steps {
+            if self.inactivity_floor_loc == self.op_count() - active_ops {
```

Comment:
I think this scenario is very rare (and not practical in a real system). I think its better to always apply the logic here (even if it is, in the worst case, over a record of all active ops).

Reply:
ok, removed. though we still need to check for out of bounds in the case the store is empty.